src/main.rs

```diff
@@ -386,10 +397,15 @@ fn static_file_handler(dest: &str, req: Request, mut res: Response) -> IoResult<
         }
     };
 
+    // strip off any querystrings so path.is_file() matches
+    // and doesn't stick index.html on the end of the path
+    // (querystrings often used for cachebusting)
+    let stripped_path = &req_path.split('?').collect::<Vec<&str>>()[0];
+
     // find the path of the file in the local system
     // (this gets rid of the '/' in `p`, so the `join()` will not replace the
     // path)
-    let path = PathBuf::from(dest).join(&req_path[1..]);
+    let path = PathBuf::from(dest).join(&stripped_path[1..]);
 
```

Comment:
Rather than `split`/`collect`, what if we
- `rfind` + `truncate`
- `rsplitn(2, "?").last().expect("always at least one element")`